openquake/commonlib/tests/sourcewriter_test.py

```diff
@@ -48,3 +37,14 @@ def test_nonparam(self):
 
     def test_alt_mfds(self):
         self.check_round_trip(ALT_MFDS)
+
+
+# deep-copied sources must be serializable
+class DeepcopyTestCase(unittest.TestCase):
```

Comment:
Please write more explicitly where things went wrong before adding the `__deepcopy__`